openupgradelib/openupgrade_merge_records.py

```diff
@@ -347,24 +348,27 @@ def _adjust_merged_values_orm(env, model_name, record_ids, target_record_id,
             elif op == 'or':
                 vals[field.name] = functools.reduce(lambda x, y: x | y, l)
         elif field.type in ('date', 'datetime'):
+            l = filter(lambda x: x is not False, l)
             op = op or 'other'
             if op == 'max':
                 vals[field.name] = max(l)
             elif op == 'min':
                 vals[field.name] = min(l)
         elif field.type == 'many2many':
+            l = filter(lambda x: x is not False, l)
             op = op or 'merge'
             if op == 'merge':
                 vals[field.name] = [(4, x.id) for x in l]
         elif field.type == 'one2many':
+            l = filter(lambda x: x is not False, l)
             op = op or 'merge'
             if op == 'merge':
                 o2m_changes += 1
                 l.write({field.inverse_name: target_record_id})
         elif field.type in ('binary', 'many2one'):
             op = op or 'merge'
             if op == 'merge':
-                l = [x for x in l if x]
+                l = filter(lambda x: x is not False, l)
```

Comment:
after thinking a bit, I think here should be `filter(lambda x: x, l)` instead, because here you don't want empty records.

Reply:
Then, Do you think is good Idea then separate binary and many2one section one if for each type? Will be something like

* at many2one use the filter(lambda x: x, l)
* at binary use the filter(lambda x: x is not False, l)

Reply:
I think it's no needed to separate them because you won't have never a 0 binary file, but feel free to separate them if you want

Reply:
well, is seems not necessary then.

Reply:
Then everything is correct now in this PR, @mreficent @zaoral ?

Reply:
No, I think here we should put the filter(lambda x: x, l) which is more strict

Reply:
please @zaoral

Reply:
@mreficent  Ready, the change you requested have been applied and pushed.

Reply:
Good, thanks